### PR TITLE
feat: drop to maintenance mode in cloud platforms if userdata is missing

### LINKED ISSUE #2780: cloud platforms: jump to maintentance mode if userdata is missing
This should only happen if userdata is missing, and as config is persisted, it should only be triggered on initial (first) boot

### PR DESCRIPTION
**Pull request by smira**
On first boot of Talos, if userdata is missing, Talos is going to drop
into maintenance mode which allows to upload config to the server via
`talosctl apply-config` command.

See also: https://github.com/talos-systems/go-retry/pull/4

Fixes #2780

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: pkg/download/download.go [hunk 1]
```diff
@@ -89,12 +106,9 @@ func Download(ctx context.Context, endpoint string, opts ...Option) (b []byte, e
...
 		}
 
-		b, err = download(req)
-		if err != nil {
-			return retry.ExpectedError(err)
-		}
+		b, err = download(req, dlOpts)
 
-		return nil
+		return err
```
Comment (Unix4ever) [inline pkg/download/download.go:111]: tiny nit: should just `return download(req, dlOpts)` work in this case?

**Review by Unix4ever — APPROVED** [review]

File: pkg/download/download.go (hunk 1 above)
Comment (smira) [reply pkg/download/download.go:111]: the reason is that `b` will not be set, function in `Retry` should return only `error`

Comment (Unix4ever) [reply pkg/download/download.go:111]: oh, right. It's nested function. I should have expanded more sources :)

File: website/content/docs/v0.7/Cloud Platforms/digitalocean.md [hunk 2]
```diff
@@ -10,16 +10,17 @@ If you need more information on Digital Ocean specifics, please see the [officia
...
 
 Using an upload method of your choice (`doctl` does not have Spaces support), upload the image to a space.
 Now, create an image using the URL of the uploaded image:
 
 ```bash
 doctl compute image create \
     --region $REGION \
-    --image-name talos-digital-ocean-tutorial \
-    --image-url https://talos-tutorial.$REGION.digitaloceanspaces.com/digital-ocean.raw.gz \
+    --image-description talos-digital-ocean-tutorial \
```
Comment (Ulexus) [inline website/content/docs/v0.7/Cloud Platforms/digitalocean.md:22]: :+1: I forgot about this; it hit me when I was getting my DO setup running.

**Review by Ulexus — APPROVED** [review]

**Comment by smira** [comment]
rebased resolving conflicts, no changes

**Pull request merged**